src/viewport/viewport.service.ts

```diff
@@ -100,72 +52,50 @@ export class ViewportService {
 
 		this.sizeType$ = this.resize$.pipe(
 			startWith(this.getViewportSize()),
-			map(x => this.calculateViewportSize(x.width)),
+			distinctUntilChanged((a, b) => a.width === b.width),
+			map(x => this.getWidthSizeInfo(x.width)),
 			distinctUntilChanged(),
 			shareReplay(1),
 		);
 	}
 
 	/**
-	 * Calculates amount of items that fits into container's width.
-	 * @param containerWidth
-	 * @param itemWidth
-	 * @returns
+	 * Returns the current viewport size
 	 */
-	calculateItemsPerRow(containerWidth: number, itemWidth: number): number {
-		if (containerWidth === 0) {
-			return 0;
-		}
-
-		if (!containerWidth && !this.windowRef.hasNative) {
-			// todo: find a way to get container width for ssr
-			containerWidth = this.viewportServerSize.get().width;
-		}
-
-		return containerWidth / itemWidth;
-	}
-
 	private getViewportSize(): ViewportSize {
 		if (!this.windowRef.hasNative) {
 			return this.viewportServerSize.get();
 		}
 
 		const ua = navigator.userAgent.toLowerCase();
-		// safari subtracts the scrollbar width
-		if (ua.indexOf("safari") !== -1 && ua.indexOf("chrome") === -1) {
+		if (ua.indexOf("safari") !== -1 && ua.indexOf("chrome") === -1) { // safari subtracts the scrollbar width
 			return {
-				width: this.document.documentElement.clientWidth,
-				height: this.document.documentElement.clientHeight,
+				width: this.windowRef.native.document.documentElement.clientWidth,
+				height: this.windowRef.native.document.documentElement.clientHeight,
```

Comment:
the document can be retrieved through window object, removed injection of document

Reply:
was gonna say leave doc, but its guarded anyway. as in general `document` should be better as it works better with SSR

Reply:
correct but since there was a condition before these that check's if the window is available, I saw that there was no need of injecting the document